engine/tests/phpunit/Elgg/ActionsServiceTest.php

```diff
@@ -1,31 +1,107 @@
 <?php
+
 namespace Elgg;
 
-class ActionsServiceTest extends \PHPUnit_Framework_TestCase {
+use Elgg\Ajax\Service;
+use Elgg\Http\ErrorResponse;
+use Elgg\Http\Input;
+use Elgg\Http\OkResponse;
+use Elgg\Http\Request;
+use Elgg\Http\ResponseFactory;
+use Elgg\I18n\Translator;
+use ElggSession;
+use PHPUnit_Framework_TestCase;
+use Symfony\Component\HttpFoundation\RedirectResponse;
+use Symfony\Component\HttpFoundation\Response;
+
+/**
+ * @group HttpService
+ * @group ActionsService
+ */
+class ActionsServiceTest extends PHPUnit_Framework_TestCase {
+
+	/**
+	 * @var ActionsService
+	 */
+	private $actions;
+
+	/**
+	 * @var Request
+	 */
+	private $request;
+
+	/**
+	 *
+	 * @var type @var \Elgg\PluginHooksService
+	 */
+	private $hooks;
 
 	public function setUp() {
+
 		$this->actionsDir = dirname(dirname(__FILE__)) . "/test_files/actions";
+
+		$session = ElggSession::getMock();
+		_elgg_services()->setValue('session', $session);
+		_elgg_services()->session->start();
+
+		$config = _elgg_testing_config();
+		_elgg_services()->setValue('config', $config);
+
+		$input = new Input();
+		_elgg_services()->setValue('input', $input);
+
+		$this->actions = new ActionsService($config, $session, _elgg_services()->crypto);
+		_elgg_services()->setValue('actions', $this->actions);
+
+		$this->request = _elgg_testing_request();
+		_elgg_services()->setValue('request', $this->request);
+
+		$this->translator = new Translator();
+		$this->translator->addTranslation('en', ['__test__' => 'Test']);
+
+		$this->hooks = new PluginHooksService();
+		$this->system_messages = new SystemMessagesService(elgg_get_session());
+	}
+
+	function createService() {
+
```

Comment:
We need a way to reset the ServiceProvider after each test case. I think overwriting hooks here is breaking the connection. The logger ends up with a different hooks instance.